pallets/did/src/lib.rs

```diff
@@ -437,10 +474,31 @@ pub mod pallet {
 				.verify_and_recover_signature(&details.encode(), &signature)
 				.map_err(Error::<T>::from)?;
 
-			let did_entry =
-				DidDetails::from_creation_details(details, account_did_auth_key).map_err(Error::<T>::from)?;
+			let did_entry = DidDetails::from_creation_details(
+				details,
+				account_did_auth_key,
+				Deposit {
+					owner: sender.clone(),
+					amount: T::Deposit::get(),
+				},
+			)
+			.map_err(Error::<T>::from)?;
+
+			// *** No Fail beyond this call ***
+			CurrencyOf::<T>::reserve(&did_entry.deposit.owner, did_entry.deposit.amount)?;
+
+			// withdraw the fee, we made sure that enough balance is available. But if this
```

Comment:
S: Start with uppercase

```suggestion
			// Withdraw the fee. We made sure that enough balance is available earlier. But if this
```